runner/src/internal.rs

```diff
@@ -103,21 +118,24 @@ impl std::fmt::Display for StatusResultValue {
 
 pub struct SingleStatusResult {
     value: StatusResultValue,
-    command: String,
     logs: String,
 }
 
-/// An execution step, which may be a single (dynamically dispatched) `Runnable`, or a collection of
-/// sub-steps.
+/// An execution step, which may be a single `Cmd`, or a collection of sub-steps.
 pub enum Step {
     Single {
         name: String,
-        runnable: Box<dyn Runnable>,
+        command: Cmd,
     },
     Multiple {
         name: String,
         steps: Vec<Step>,
     },
+    WithBackground {
+        name: String,
+        background: Cmd,
+        foreground: Box<Step>,
+    },
```

Comment:
Do you think these could possibly be merged? I'm of course very new to this, so bear with me, but it seems to me as if "Multiple" is pretty much equivalent to a "WithBackground" with the background process being optional. 

Also WithBackground is a bit of a mix of both Single & Multiple, since I assume it will be used to execute both examples with just a single client, and with multiple ones.

Feel free to dismiss this if this doesn't make sense, but it seems to be as if one could also do

```rust
pub struct Step {
    name: String,
    execute: enum Execute {
        Cmd, Vec<Step>
    },
    background: Option<Cmd>
}
```

Reply:
That's actually a great idea! I got so used to the current structure that I missed this shared structure, thanks for pointing out! I'll do that in a follow up PR.

Reply:
Awesome! It seemed like the current structure was more informed by the previous requirements than the ones intro'ed by this PR.  I'm happy to hear that, and that apparently my experience refactoring Js data structures is somewhat transferrable. :))